core/executor.py

```diff
@@ -422,6 +424,7 @@ def RunPipeline(self, node, status_out):
     def RunSubshell(self, node):
         # type: (command_t) -> int
         p = self._MakeProcess(node)
+        p.SetBackground()
```

Comment:
Hm why is a subshell in the background?

Does "in background" mean "not part of the single PGID that has terminal control" ?   


Or something related but similar?  we should be precise about the terminology

Reply:
Ah yes. In my head "in the background" is synonymous with "cannot use the terminal"

Perhaps we should rename again from `in_background` to something like `has_tty` to avoid confusion